test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc021.java

```diff
@@ -0,0 +1,99 @@
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.jvmti.tests.redefineClasses;
+
+import com.ibm.jvmti.tests.util.Util;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.lang.annotation.Annotation;
+
+public class rc021 {
+	public static native boolean redefineClass(Class name, int classBytesSize,
+			byte[] classBytes);
+
+	final static int ITERATIONS = 100;
+	static boolean result = true;
+
+	public boolean testAnnotationCacheAfterRedefine() {
+		try {
+			final class AnnotationFetch implements Runnable {
+				@Override
+				public void run() {
+					try {
+						Class original = rc021_testAnnotationCache_O.class;
+
+						Annotation[] ann = original.getDeclaredAnnotations();
+						if (ann.length == 0) {
+							System.out.println("No annotations");
+						} else {
+							System.out.println("Annotations:");
+						}
+						for (Annotation a : ann) {
+							System.out.println(a);
+						}
+					} catch (Throwable t) {
+						System.err.println("Exception during annotation fetch:");
+						t.printStackTrace();
+						result = false;
+					}
+				}
+			};
+			final class Redef implements Runnable {
+				@Override
+				public void run() {
+					boolean redefined = Util.redefineClass(rc021.class, rc021_testAnnotationCache_O.class, rc021_testAnnotationCache_R.class);
+					if (!redefined) result = false;
+					redefined = Util.redefineClass(rc021.class, rc021_testAnnotationCache_R.class, rc021_testAnnotationCache_O.class);
+					if (!redefined) result = false;
+				}
+			};
+
+			// for (int i = 0; i < ITERATIONS; i++) {
+			// 	Thread t1 = new Thread(new AnnotationFetch());
+			// 	Thread t2 = new Thread(new Redef());
+			// 	t2.start();
+			// 	t1.start();
+			// 	t2.join();
+			// 	t1.join();
+			// }
+
+			ExecutorService executor = Executors.newCachedThreadPool();
+			for (int i = 0; i < ITERATIONS; i++) {
+				executor.submit(new AnnotationFetch());
+				executor.submit(new Redef());
+			}
+
+			executor.shutdown();
```

Comment:
Add a comment that invoking shutdown also executes the submitted threads before shutting down.